modules/siddhi-launcher/src/main/java/org/wso2/siddhi/sdk/launcher/Main.java

```diff
@@ -0,0 +1,134 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.siddhi.sdk.launcher;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.MissingCommandException;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParameterException;
+import com.beust.jcommander.Parameters;
+import org.apache.log4j.Logger;
+import org.wso2.siddhi.sdk.launcher.exception.SLauncherException;
+import org.wso2.siddhi.sdk.launcher.util.Constants;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * This class executes a siddhi program.
+ */
+public class Main {
+
+    /**
+     * Logger to log events.
+     */
+    private static final Logger log = Logger.getLogger(Main.class);
+
+    private static final String JC_UNKNOWN_OPTION_PREFIX = "Unknown option:";
+    private static final String JC_EXPECTED_A_VALUE_AFTER_PARAMETER_PREFIX = "Expected a value after parameter";
+
+    public static void main(String... args) {
+        try {
+            log.debug("jkhbv");
+            Optional<SLauncherCmd> optionalInvokedCmd = getInvokedCmd(args);
+            optionalInvokedCmd.ifPresent(SLauncherCmd::execute);
+        } catch (SLauncherException e) {
+            log.error(e.getMessage());
+            Runtime.getRuntime().exit(1);
+        } catch (Throwable e) {
+            String msg = "siddhi: " + e.toString();
+            log.error(msg);
+            Runtime.getRuntime().exit(1);
+        }
+    }
+
+    private static Optional<SLauncherCmd> getInvokedCmd(String... args) throws SLauncherException {
+        try {
+            // Run command
+            RunCmd runCmd = new RunCmd();
+            JCommander cmdParser = new JCommander(runCmd);
+            cmdParser.setProgramName("siddhi");
+            cmdParser.parse(args);
+            String parsedCmdName = cmdParser.getParsedCommand();
+            // User has not specified a command. Therefore returning the main command
+            // which simply prints usage information.
+            if (parsedCmdName == null) {
+                return Optional.of(runCmd);
+            }
+            Map<String, JCommander> commanderMap = cmdParser.getCommands();
+            return Optional.of((SLauncherCmd) commanderMap.get(parsedCmdName).getObjects().get(0));
+        } catch (MissingCommandException e) {
+            String errorMsg = "unknown command '" + e.getUnknownCommand() + "'";
+            throw LauncherUtils.createUsageException(errorMsg);
+        } catch (ParameterException e) {
+            String msg = e.getMessage();
+            if (msg == null) {
+                throw LauncherUtils.createUsageException("internal error occurred");
+            } else if (msg.startsWith(JC_UNKNOWN_OPTION_PREFIX)) {
```

Comment:
Isn't there a better way to ID each scenario that String manipulations?

Reply:
This is the usual way of handling exceptions thrown by JCommander.